html/includes/api_functions.inc.php

```diff
@@ -822,11 +845,37 @@ function get_ip_addresses()
         check_device_permission($device_id);
         $ipv4   = dbFetchRows("SELECT `ipv4_addresses`.* FROM `ipv4_addresses` JOIN `ports` ON `ports`.`port_id`=`ipv4_addresses`.`port_id` WHERE `ports`.`device_id` = ? AND `deleted` = 0", array($device_id));
         $ipv6   = dbFetchRows("SELECT `ipv6_addresses`.* FROM `ipv6_addresses` JOIN `ports` ON `ports`.`port_id`=`ipv6_addresses`.`port_id` WHERE `ports`.`device_id` = ? AND `deleted` = 0", array($device_id));
+        $ip_addresses_count = count(array_merge($ipv4, $ipv6));
+        if (!is_numeric($ip_addresses_count)) {
+            api_error(500, 'Error retrieving IP Addresses');
+        }
+        if ($ip_addresses_count == 0) {
```

Comment:
This just means it might have no IPs. I wouldn't indicate the device doesn't exist here.

Reply:
Will fix in followup